hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileReaderImpl.java

```diff
@@ -168,14 +168,26 @@ public int seekTo(Key key) throws IOException {
     if (compareCurrent == 0) {
       return SEEK_TO_FOUND;
     }
-    if (!isAtFirstKey()) {
-      // For backward seekTo after the first key, throw exception
-      throw new IllegalStateException(
-          "The current lookup key is less than the current position of the cursor, "
-              + "i.e., backward seekTo, which is not supported and should be avoided. "
-              + "key=" + key + " cursor=" + cursor);
+    // compareCurrent < 0, i.e., the lookup key is lexicographically smaller than
```

Comment:
this java doc is confusing when read along w/ the `if` condition, where we check for >=0

Reply:
`key.compareTo(currentDataBlockEntry.get().getFirstKey()) >= 0` is different.  This remaining part meets the condition of `compareCurrent < 0` because previously `compareCurrent > 0` and ``compareCurrent == 0` cases are handled.

Reply:
Improved the docs to avoid confusion